writings/2021-12-12-the-importance-of-the-most-profitable-block.mdx

```diff
@@ -0,0 +1,86 @@
+
+First, searchers submit individual bundles and while many of these bundles are conflicting (e.g. they bid for the same opportunity) many bundles can be merged together without a problem. Figuring out the ordering and number of bundles that optimizes a miner’s profits is a very difficult problem and is broadly referred to as “bundle merging.”
+
+To merge bundles together they first need to be simulated alone to understand which bundle should be placed first according to the gas price of those bundles. Then bundles must be simulated together one after another to understand if they conflict in some way. Lastly for each additional bundle that is being attempted to be merged an additional and parallel computing process needs to be created. For example, if a miner wants to merge a maximum number of 3 bundles together they will have 4 “workers”: one creating a block without bundles, another creating a block with 1 bundle, a third with 2 bundles, and a fourth with 3 bundles. Then these four blocks are compared and the most profitable block is chosen.
+
+If the details of the above are lost on you that’s fine, just remember that bundle merging is a computationally complicated process. And sometimes there can be tens of thousands of bundles for a single block! Flashbots’ MEV-Geth handles bundle merging for miners out of the box, but the [computational complexity of merging](https://twitter.com/bertcmiller/status/1442548316735148037) places a cap on the number of bundles that can be included in a given block. The median miner will merge a maximum of 3 bundles per block, but many blocks have had more than 3 bundles that are profitable to include.
+
+Moreover, Flashbots Alpha auctions off block space by optimizing for bundles with the highest gas price. Unfortunately this is not the same as optimizing for overall bundle profit. Consider the following two bundles:
+* Bundle A: 100,000 gas with miner payment of 1 ETH and gas price of 10,000 GWEI
+* Bundle B: 1,000,000 gas with miner payment of 9.9 ETH and gas price of 9,900 GWEI
+
+Bundle A will win the Flashbots Auction because it has a higher gas price despite bundle B being more profitable overall for miners.
+
+The reason this design decision was made is because optimizing bundles for gas price is computationally simpler than optimizing for overall profit. However this opens up a dimension for competition: if a miner has a proprietary solution that allows them to optimize bundles for overall profit they could extract more MEV than their peers in some cases. Similarly, being able to include more bundles per block would also lead to more MEV extraction.
```

Comment:
First sentence: not sure I agree. I believe the reason we did this was to prevent "bundle stuffing", i.e. adding unrelated txs to pump the bundle price. not sure why we're still normalizing by total gas after we explicitly subtract the contributions of mempool txs tho, I don't think we should (and we ultimately won't when moving to full blocks)